docs2/site/docs/getting-started/directives.md

```diff
@@ -71,6 +71,203 @@ when the same directive should be used with different arguments at a single loca
+    }
+}
+```
+
+Applying `@length` directive to a field argument.
+
+```csharp
+public class Query : ObjectGraphType
+{
+    public Query()
+    {
+        Field<Human>(
+            "human",
+            arguments: new QueryArguments(
+                new QueryArgument<IdGraphType>
+                {
+                    Name = "id"
+                }
+                .ApplyDirective("length", "min", 2, "max", 5)
+            ));
+    }
+}
+```
+
+# How do directives work
+
+So you have defined a directive. Then you have applied (or not in case of client-side directive) this
+directive to the required locations in your schema. What's next? So far, all you have done is set some
+meta information, and there is still no code anywhere that is responsible for the actions of the added
+directive. The next step is to define a class that will customize the schema using the information
+provided by the applied directive. This class should implement `ISchemaNodeVisitor` interface.
+
+Let's imagine an `@upper` directive.
+
+```csharp
+public class UpperDirective : DirectiveGraphType
+{
+    public UpperDirective()
+        : base("upper", DirectiveLocation.FieldDefinition)
+    {
+        Description = "Converts the value of string fields to uppercase.";
+    }
+}
+```
+
+To make this directive work, you need to write a class like the following (`BaseSchemaNodeVisitor` is just
+a base class implementing `ISchemaNodeVisitor` with empty `virtual` methods, it does nothing).
+
+```csharp
+public class UppercaseDirectiveVisitor : BaseSchemaNodeVisitor
+{
+    public override void VisitFieldDefinition(FieldType field, ISchema schema)
+    {
+        if (field.HasAppliedDirectives() && field.GetAppliedDirectives().Find("upper") != null)
+        {
+            var inner = field.Resolver ?? NameFieldResolver.Instance;
+            field.Resolver = new FuncFieldResolver<object>(context =>
+            {
+                object result = inner.Resolve(context);
+
+                return result switch
+                {
+                    string str => str?.ToUpperInvariant(),
+                    Task<string> task => Task.FromResult(task.GetAwaiter().GetResult()?.ToUpperInvariant()),
+                    _ => result
+                };
+            });
+        }
+    }
+}
+```
+
+And then register schema visitor within your schema just like you did to register the directive.
+
+```csharp
+public class MySchema : Schema
+{
+    public MySchema()
+    {
+        RegisterVisitor(new UppercaseDirectiveVisitor());
+        // or
+        RegisterVisitor(typeof(UppercaseDirectiveVisitor));
+    }
+}
+```
+
+Note that a schema visitor, unlike a directive, can be registered not only as an instance but also as
+a type. In this case, when initializing the schema, schema visitor will be created according to how
+you configure the DI container. In other words, schema visitors support dependency injection.
```

Comment:
Add note about how schema visitors, when created via DI, have same lifetime as schema, and refer to dependency injection notes in middleware guide.

Reply:
Done. Also see #2299